### PR TITLE
Cleanup rustdoc themes

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
This PR continues our work to simplify the rustdoc themes by relying more on CSS variables. Interestingly enough, this time it allowed me to realize that we were having a lot of different colors for borders even though the difference is unnoticeable. I used this opportunity to unify them.

Follow-up of #98460.

The live demo is [here](https://rustdoc.crud.net/imperio/cleanup-themes/std/index.html).

r? @jsha

### CODE REVIEW

File: src/librustdoc/html/static/css/rustdoc.css [hunk 1]
```diff
@@ -693,8 +696,8 @@ h2.location a {
 	flex-grow: 1;
 	margin: 0px;
 	padding: 0px;
-	overflow-wrap: break-word;
```
Comment (jsha) [inline src/librustdoc/html/static/css/rustdoc.css:696]: The use of `break-word` followed by `anywhere` is intentional. See https://github.com/rust-lang/rust/blob/a39bdb1d6b9eaf23f2636baee0949d67890abcd8/src/librustdoc/html/static/css/rustdoc.css#L107-L112.

File: src/librustdoc/html/static/css/rustdoc.css [hunk 2]
```diff
@@ -974,7 +977,10 @@ table,
 	max-width: 100%;
 	/* contents can overflow because of max-width limit, then show ellipsis */
 	text-overflow: ellipsis;
-	border: 1px solid;
+	/* Without the `!important`, the border-color is ignored for `<select>`...
+	   It cannot be in the group above because `.search-input` has a different border color on
+	   hover. */
+	border: 1px solid var(--border-color) !important;
```
Comment (jsha) [inline src/librustdoc/html/static/css/rustdoc.css:983]: I don't understand this comment. I think `border: 1px solid var(--border-color)` should work. 

Are you familiar with how [CSS Specificity works](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity)? Most instances of `!important` can be better solved by applying the principles of specificity.

Also, when you say "cannot be in the group above", (a) you should say "in the rule above" or "in the ruleset above" (https://developer.mozilla.org/en-US/docs/Web/CSS/Syntax#css_rulesets). Using precise language helps clarify. Also: are you talking about the ruleset with the selector `#crate-search-div`? I don't see how that one is relevant.

File: src/librustdoc/html/static/css/rustdoc.css [hunk 3]
```diff
@@ -1520,17 +1527,20 @@ pre.rust {
 #settings-menu, #help-button {
 	margin-left: 4px;
 	outline: none;
+	background-color: var(--search-elements-background-color);
```
Comment (jsha) [inline src/librustdoc/html/static/css/rustdoc.css:1530]: I think `--search-elements-background-color` is a misleading name for this variable. The settings menu button and the help button are only related to the search button by location, not by semantics. Perhaps this should be `--button-background-color`?

File: src/librustdoc/html/static/css/themes/ayu.css [hunk 4]
```diff
@@ -336,7 +312,7 @@ pre.ignore:hover, .information:hover + pre.ignore {
 .tooltip::after {
 	background-color: #314559;
 	color: #c5c5c5;
-	border: 1px solid #5c6773;
+	border: 1px solid var(--border-color);
```
Comment (jsha) [inline src/librustdoc/html/static/css/themes/ayu.css:315]: I think we should no longer need to set the `border` property in stylesheets at all, right?

**Review by jsha — CHANGES_REQUESTED** [review]

File: src/librustdoc/html/static/css/rustdoc.css (hunk 1 above)
Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/rustdoc.css:696]: Oh interesting! I'll put the comment here too then.

File: src/librustdoc/html/static/css/rustdoc.css (hunk 3 above)
Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/rustdoc.css:1530]: They are in the "search-nav" so I thought it made sense but `--button-background-color` sounds better indeed.

File: src/librustdoc/html/static/css/themes/ayu.css (hunk 4 above)
Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/themes/ayu.css:315]: You mean we could write:

```css
body {
    border-color: var(--border-color);
}
```

?

Comment (jsha) [reply src/librustdoc/html/static/css/themes/ayu.css:315]: No, I mean there should be a rule in rustdoc.css that says `.tooltip::after { border: 1px solid var(--border-color) }`, and this property should be deleted from the ayu theme.

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/themes/ayu.css:315]: Oh sorry, I didn't pay attention to the file. You're right!

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/themes/ayu.css:315]: Actually, `ayu` is a bit on its own here: the other themes don't have a border around the tooltip (you can see them in https://doc.rust-lang.org/nightly/std/string/struct.String.html#deref). A preference over what should be done?

**Comment by GuillaumeGomez** [comment]
I also updated the online demo.

File: src/librustdoc/html/static/css/themes/ayu.css (hunk 4 above)
Comment (jsha) [reply src/librustdoc/html/static/css/themes/ayu.css:315]: I think all the themes should have a thin border around the tooltip. This is consistent with what we do for the help and settings overlays, and for the notable traits overlay.

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/themes/ayu.css:315]: Good point. I'll add the border color on all tooltips then.

**Comment by GuillaumeGomez** [comment]
Added the border everywhere and removed it from ayu theme.

File: src/librustdoc/html/static/css/rustdoc.css [hunk 5]
```diff
@@ -138,6 +138,9 @@ h1, h2, h3, h4 {
 .docblock h3, .docblock h4, h5, h6 {
 	margin: 15px 0 5px 0;
 }
+.docblock table td, .docblock table th {
+	border-color: var(--border-color);
```
Comment (jsha) [inline src/librustdoc/html/static/css/rustdoc.css:142]: This should be combined with the rules that set the border for these elements in the first place.

```css
.docblock table td {
	padding: .5em;
	border: 1px dashed;
}

.docblock table th {
	padding: .5em;
	text-align: left;
	border: 1px solid;
}
```

File: src/librustdoc/html/static/css/rustdoc.css [hunk 6]
```diff
@@ -1520,17 +1530,20 @@ pre.rust {
...
 	height: 34px;
+	background-color: var(--main-background-color);
 }
 #settings-menu > a, #help-button > button, #copy-path {
 	padding: 5px;
 	width: 33px;
 	border: 1px solid;
 	border-radius: 2px;
 	cursor: pointer;
+	border-color: var(--border-color);
```
Comment (jsha) [inline src/librustdoc/html/static/css/rustdoc.css:1546]: This should be combined with the `border` property on line 1543 above.

File: src/librustdoc/html/static/css/rustdoc.css [hunk 7]
```diff
@@ -1539,6 +1552,7 @@ pre.rust {
 	padding: 5px;
 	height: 100%;
 	display: block;
+	background-color: var(--button-background-color);
```
Comment (jsha) [inline src/librustdoc/html/static/css/rustdoc.css:1555]: I think this is not needed. It's redundant with the ruleset at line 1540.

File: src/librustdoc/html/static/css/rustdoc.css [hunk 8]
```diff
@@ -840,11 +848,11 @@ nav.main {
 	text-align: center;
 }
 nav.main .current {
```
Comment (jsha) [inline src/librustdoc/html/static/css/rustdoc.css:850]: I think we can remove this ruleset entirely. We don't currently generate a `nav.main .current` AFAICT.

**Review by jsha — CHANGES_REQUESTED** [review]

File: src/librustdoc/html/static/css/rustdoc.css (hunk 8 above)
Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/rustdoc.css:850]: We don't seem to have a `nav.main` indeed (but we do have `.current`) so this rule is unused.

File: src/librustdoc/html/static/css/rustdoc.css (hunk 6 above)
Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/rustdoc.css:1546]: I'm quite ashamed of this one...

File: src/librustdoc/html/static/css/rustdoc.css (hunk 5 above)
Comment (GuillaumeGomez) [reply src/librustdoc/html/static/css/rustdoc.css:142]: Yes absolutely. Don't know how I didn't think about it...

**Comment by GuillaumeGomez** [comment]
Applied suggestions and updated live demo.

**Review by jsha — APPROVED** [review]

**Pull request merged**